### PR TITLE
Deprecate X and P

### PR DESCRIPTION
**Pull request by lillian542**
Deprecate the CV observables `qml.X` and `qml.P`, renaming them as `qml.QuadX` and `qml.QuadP`. This will free up `qml.X` for `qml.PauliX`.

This PR:

- renames the classes
- updates tests and documentation to use the new names
- `qml.X` and `qml.P` still exist as aliases, but raise a warning 
- tests that the warning is raised
- updates `default_gaussian` device to accept both `X`/`P` and `QuadX`/`QuadP`

### CODE REVIEW

File: pennylane/ops/cv.py [hunk 1]
```diff
@@ -1164,7 +1165,40 @@ def _heisenberg_rep(p):
         return np.array([0, 1, 0])
 
 
-class P(CVObservable):
+class X(QuadX):
```
Comment (lillian542) [inline pennylane/ops/cv.py:1168]: Is this a suitable way of keeping `qml.X` as an alias for now?

File: doc/development/deprecations.rst [hunk 2]
```diff
@@ -55,6 +55,12 @@ Pending deprecations
 
   - Deprecated in v0.31
 
+* The CV observables ``qml.X`` and ``qml.P`` have been deprecated, and usage will now
+  raise a warning. Please use ``qml.QuadX`` and ``qml.QuadP`` instead.
+
+  - Deprecated in v0.32
+  - Will be removed in v0.33
```
Comment (lillian542) [inline doc/development/deprecations.rst:62]: Is removal in v0.33 accurate?

**Comment by lillian542** [comment]
[sc-39024]

File: doc/development/deprecations.rst (hunk 2 above)
Comment (trbromley) [reply doc/development/deprecations.rst:62]: This should be fine, since most of our users are qubit-focused.

File: pennylane/ops/cv.py [hunk 3]
```diff
@@ -1164,7 +1165,40 @@ def _heisenberg_rep(p):
...
+    r"""
+    This class is deprecated and will be removed in a future version of PennyLane.
```
Comment (trbromley) [inline pennylane/ops/cv.py:1170]: We could mention the target removal

File: doc/code/qml.rst [hunk 4]
```diff
@@ -6,4 +6,4 @@ qml
 .. automodapi:: pennylane
     :no-heading:
     :include-all-objects:
-    :skip: iter_entry_points, Version, SimpleSpec, plugin_devices, plugin_converters, default_config, reload
+    :skip: iter_entry_points, Version, SimpleSpec, plugin_devices, plugin_converters, default_config, reload, X, P
```
Comment (trbromley) [inline doc/code/qml.rst:9]: Does this remove it completely from the docs?

Comment (lillian542) [reply doc/code/qml.rst:9]: It removes it from the qml page, but it's possible I missed something somewhere. I also had to remove it separately from the Operators intro page. If it shows up somewhere other than those two pages, its still there. But I didn't find it anywhere else. It is admittedly harder to search for than most things 😆

File: pennylane/ops/cv.py (hunk 3 above)
Comment (lillian542) [inline pennylane/ops/cv.py:1170]:
```suggestion
    This class is deprecated and will be removed in PennyLane v0.33.
```

File: pennylane/ops/cv.py [hunk 5]
```diff
@@ -1197,6 +1231,39 @@ def _heisenberg_rep(p):
         return np.array([0, 0, 1])
 
 
+class P(QuadP):
+    r"""
+    This class is deprecated and will be removed in a future version of PennyLane.
```
Comment (lillian542) [inline pennylane/ops/cv.py:1236]:
```suggestion
    This class is deprecated and will be removed in PennyLane v0.33.
```

**Review by mudit2812 — APPROVED** [review]
Thanks Lillian! I left one non-blocking suggestion, take it or leave it. Otherwise LGTM 🚀

**Review by timmysilv — APPROVED** [review]

File: doc/code/qml.rst (hunk 4 above)
Comment (trbromley) [reply doc/code/qml.rst:9]: Thanks! To clarify, I don't think we should remove it from the docs at this stage - users should be able to see `X` and `P` in the docs and read the warning about its name changing. Indeed, that's what we already have [here](https://github.com/PennyLaneAI/pennylane/pull/4330/files#diff-beac1f4626960fd49cbd9c6fef0956bc07957e80117c1081221e20e4ce0c0c71R1170), we just need to show it.

File: pennylane/ops/cv.py [hunk 6]
```diff
@@ -1164,7 +1165,40 @@ def _heisenberg_rep(p):
         return np.array([0, 1, 0])
 
 
-class P(CVObservable):
+class X(QuadX):
+    r"""
+    This class is deprecated and will be removed in PennyLane v0.33.
+    Please use qml.QuadX instead.
```
Comment (trbromley) [inline pennylane/ops/cv.py:1171]: Can we update this to be a warning, to be just above `Details`, and to link dynamically to `qml.QuadX`. The same applies to `QuadP`.

**Review by trbromley — CHANGES_REQUESTED** [review]

File: doc/code/qml.rst (hunk 4 above)
Comment (lillian542) [reply doc/code/qml.rst:9]: Sounds good, I'll put it back!

**Review by timmysilv — APPROVED** [review]
agreed with Tom's suggestion, still looks good! I noticed the base class for `X` and `P` doesn't hyperlink, but that's not the biggest deal (plus, the warning blob does)

**Pull request merged**